src/index.ts

```diff
@@ -854,6 +870,57 @@ export abstract class PinejsClientCore<PinejsClient> {
 		}
 	}
 
+	private canRetryDefaultHandler(err: any) {
+		const code = err?.statusCode;
+		return code == null || code === 429 || (code >= 500 && code < 600);
+	}
+
+	protected async callWithRetry<T>(
+		fnCall: () => Promise<T>,
+		retry: RetryParameters,
+	): Promise<T> {
+		if (!retry) {
+			return await fnCall();
+		}
+
+		const defaultRetry = this.retry || {};
+		const minDelayMs = retry.minDelayMs ?? defaultRetry.minDelayMs ?? 0;
+		const maxDelayMs = retry.maxDelayMs ?? defaultRetry.maxDelayMs ?? 0;
+		const maxAttempts = retry.maxAttempts ?? defaultRetry.maxAttempts ?? 0;
+
+		if (
+			minDelayMs <= 0 ||
+			maxDelayMs <= 0 ||
+			maxAttempts <= 0 ||
+			minDelayMs > maxDelayMs
+		) {
+			throw new Error('pinejs-client invalid retry parameters');
+		}
+
+		const onRetryHandler = retry.onRetry ?? defaultRetry.onRetry;
+
+		const attempt = 1;
```

Comment:
This probably shouldn't be able to be a `const` :P

Reply:
hahah.... ooops.